### PR TITLE
Minor clean up of datetime and other classes (#198)

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: MitchellGale-BitQuill <mitchellg@bitquilltech.com>
Signed-off-by: Yury-Fridlyand <yury.fridlyand@improving.com>
Co-authored-by: Yury-Fridlyand <yury.fridlyand@improving.com>

### Description
Clean up lexer to make future changes easier. Make changes to documentation and code to make cleaner. 

* Groom ANTLR grammar files to make future changes easier.
* Fix `EqualsAndHashCode` compilation warnings
>
```
Generating equals/hashCode implementation but without a call to superclass, even though this class does not extend java.lang.Object. If this is intentional, add '@EqualsAndHashCode(callSuper=false)' to your type.
```
* Move `ExprTimestampValue.ZONE` constant to `DateTimeUtils`
* Move `ExprDatetimeValue` formatter to `DateTimeFormatters`
* Sort alphabetically functions in `DateTimeFunction` and `MathematicalFunction`
* Replace `FunctionDSL` import by static imports to simplify code
* Remove tests for `NULL` and `MISSING` values
* Add validation for non-zero divider
* Remove `Environment` mock from some unit tests, because it is not needed anymore
* Groom docs to eliminate `rst` validator warnings
* Add doctests for some functions
* Make IT for `now`-like functions truly parametrized
* Rename some PPL grammar entities to be consistent with SQL ones
* Change doctest cluster type to `archive` which has built-in modules - this requires for some added doctests

See team review and discussion in https://github.com/Bit-Quill/opensearch-project-sql/pull/198
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by MaxKsyunz — DISMISSED** [review]

File: doctest/build.gradle [hunk 1]
```diff
@@ -139,7 +139,7 @@ testClusters {
         }))
         */
         plugin ':opensearch-sql-plugin'
-        testDistribution = 'integ_test'
+        testDistribution = 'archive'
```
Comment (acarbonetto) [inline doctest/build.gradle:142]: need this for painless

**Review by acarbonetto — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Rebased

**Comment by Yury-Fridlyand** [comment]
Rebased again

**Review by dai-chen — APPROVED** [review]
Thanks for the refactoring!

**Review by penghuo — DISMISSED** [review]

**Comment by Yury-Fridlyand** [comment]
Rebased again, please re-review

**Review by dai-chen — DISMISSED** [review]

**Review by acarbonetto — DISMISSED** [review]

**Review by dai-chen — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

**Pull request merged**